### DISCUSSION TITLE
Adding `pseudocode.js` to header doesn't work

### DISCUSSION THREAD

**Discussion by mendax1234** [academicpages/academicpages.github.io#2142] [Q&A]
Hi, I try to follow [this guide](https://rpubs.com/cderv/pseudocodejs) to add `pseudocode.js` to my head in the site. What I have done is add the code below to my `head/custom.html`
```html
<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [['$','$'], ['\\(','\\)']],
            displayMath: [['$$','$$'], ['\\[','\\]']],
            processEscapes: true,
            processEnvironments: true,
        }
    });
</script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/pseudocode@latest/build/pseudocode.min.css">
<script src="https://cdn.jsdelivr.net/npm/pseudocode@latest/build/pseudocode.min.js"></script>
```
And I don't add this code in my R markdown file
```yaml
output: 
  html_document:
    self_contained: false
    includes:
      in_header: head.html
```

But when I try to render the code below, it doesn't work.

```{.algorithm}
% This quicksort algorithm is extracted from Chapter 7, Introduction to Algorithms (3rd edition)
\begin{algorithm}
\caption{Quicksort}
\begin{algorithmic}
\PROCEDURE{Quicksort}{$A, p, r$}
    \IF{$p < r$} 
        \STATE $q = $ \CALL{Partition}{$A, p, r$}
        \STATE \CALL{Quicksort}{$A, p, q - 1$}
        \STATE \CALL{Quicksort}{$A, q + 1, r$}
    \ENDIF
\ENDPROCEDURE
\PROCEDURE{Partition}{$A, p, r$}
    \STATE $x = A[r]$
    \STATE $i = p - 1$
    \FOR{$j = p$ \TO $r - 1$}
        \IF{$A[j] < x$}
            \STATE $i = i + 1$
            \STATE exchange
            $A[i]$ with $A[j]$
        \ENDIF
        \STATE exchange $A[i]$ with $A[r]$
    \ENDFOR
\ENDPROCEDURE
\end{algorithmic}
\end{algorithm}
```

Can someone help me with it!

Thanks!

**Comment by rjzupkoii** [comment]
Do you know if that plug-in supports Markdown rendering other than R Markdown? The Academic Pages template uses [karmadown](https://kramdown.gettalong.org/), and there are enough differences between the various Markdown engines that the plug-ins aren't always compatible.

**Reply by mendax1234** [reply]
@rjzupkoii I am not sure whether it supports markdown or not. But I have successfully achieved this function on the academicpages. And here are my steps
1. Add the following code in`_includes/head/custom.html`
```html
<!-- Support for pseudocode.js -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/pseudocode@latest/build/pseudocode.min.css">
<script src="https://cdn.jsdelivr.net/npm/pseudocode@latest/build/pseudocode.min.js"></script>
```
2. Add the following code in `_includes/scripts.html`
```html
<script>
    pseudocode.renderClass("pseudocode");
</script>
```
3. Then we can write pseudocode in academicpages
4. Another thing I change is the MathJax3.0, I add the following to my `_includes/head/custom.html`
```html
<script>
    MathJax = {
        tex: {
            inlineMath: [['$','$'], ['\\(','\\)']],
            displayMath: [['$$','$$'], ['\\[','\\]']],
            processEscapes: true,
            processEnvironments: true,
        }
    }
</script>
```

For my solution, I don't know whether my step 2 will affect my other content or not. I think it is okay. Can you help me check it?
Thank you very much!

Besides, if you think it is a good function, I will try to create a PR.

**Reply by mendax1234** [reply]
@rjzupkoii There is bug now and that's the rendering doesn't work on mobile site. Can I confirm with you that whether the `script.html` will be included in the mobile site?
Thanks!

**Reply by rjzupkoii** [reply]
@mendax1234 Excellent work! Everything in `script.html` is included in the mobile site - remember that the template uses responsive design so it's detecting the resolution when deciding to display in a mobile view or not. Hard to say what would cause rendering issues without actually taking a look at the code itself and checking the console (under More Tools > Developer Tools > Console in Chrome) to see if there are any messages present.

**Reply by mendax1234** [reply]
> @mendax1234 Excellent work! Everything in `script.html` is included in the mobile site - remember that the template uses responsive design so it's detecting the resolution when deciding to display in a mobile view or not. Hard to say what would cause rendering issues without actually taking a look at the code itself and checking the console (under More Tools > Developer Tools > Console in Chrome) to see if there are any messages present.

@rjzupkoii Thanks for your advice! But seems that there is no developer tools in mobile version chrome, so it's hard for me to debug. I think you are right, may be it is because of the responsive design of this theme, not the rendering. If so, can you give me some hints on solving this problem? Since I am pretty new to the responsive design and don't know how it's implemented in this awesome project.

To reproduce the bug on mobile version, I think just need to add the code in my above reply and then add an example pseudocode in one of the posts and then use the mobile to see the post.

Thanks for any help!

**Reply by rjzupkoii** [reply]
@mendax1234 You can actually use desktop Chrome to do mobile development! If you open the Developer Tools there's a small icon that allows you to [open the Device Toolbar](https://developer.chrome.com/docs/devtools/device-mode) which then allows you to emulate a mobile device.

**Reply by mendax1234** [reply]
> @mendax1234 You can actually use desktop Chrome to do mobile development! If you open the Developer Tools there's a small icon that allows you to [open the Device Toolbar](https://developer.chrome.com/docs/devtools/device-mode) which then allows you to emulate a mobile device.

![image](https://github.com/academicpages/academicpages.github.io/assets/62934294/7b8e0acd-61a6-4818-a70c-426fd2dff362)
@rjzupkoii Thanks. I see I see. But seems that based on the simulation, the pseudocode.js should work find on mobile. That's strange.
